### DISCUSSION TITLE
Proposal for Exercise in Unit 2

### DISCUSSION THREAD

**Discussion by michaelshekasta** [huggingface/computer-vision-course#295] [02. Convolutional Neural Networks]
Greetings everyone,

I'd like to propose a slight adjustment to our approach for Unit 2. Rather than focusing solely on assignments and grades, I believe it would be beneficial to emphasize tasks that foster a deeper understanding among learners.

In my opinion, this unit could be structured without formal grading, but instead, we could introduce a task that encourages hands-on exploration. One idea I had is to delve into a simple yet insightful task, such as utilizing the "minst" dataset, and experimenting with different CNN architectures. For instance, we could compare the performance of a single CNN with numerous kernels against the effectiveness of employing two CNNs sequentially, among other variations.

I'm eager to gather your feedback and insights regarding this proposal.

Best regards,
Michael

**Comment by michaelshekasta** [comment]
For graded assignment - We can utilize a well-established dataset like ImageNet and evaluate the outcomes using the test dataset. The results can be showcased on a leaderboard hosted by Gradio.

Furthermore, I suggest incorporating additional components such as labs. In my opinion, delving into the contrasting approaches—widely covering single CNN with numerous kernels versus depth (employing two smaller CNNs sequentially)—will provide valuable insights.

What do you think?
@johko @mmhamdy @ATaylorAerospace @merveenoyan

**Reply by mmhamdy** [reply]
I like the idea! 😀 but I think it would be better to keep the exercise after each unit simple. We can still have a lab with as many creative exercises and a leaderboard as an optional activity at the end of the course covering as many topics as you like. But this is just a suggestion, let's see what others think.

**Reply by michaelshekasta** [reply]
@mmhamdy Let's talk about it all in [one place](https://github.com/johko/computer-vision-course/discussions/290#discussioncomment-9316816) instead of two
